pallets/runtime/tests/src/relayer_test.rs

```diff
@@ -0,0 +1,128 @@
+use super::{
+    storage::{get_last_auth_id, TestStorage, User},
+    ExtBuilder,
+};
+use frame_support::{assert_noop, assert_ok, StorageMap};
+use polymesh_common_utilities::traits::transaction_payment::CddAndFeeDetails;
+use polymesh_common_utilities::Context;
+use polymesh_primitives::{Signatory, TransactionError};
+use polymesh_runtime_develop::{fee_details::CddHandler, runtime::Call};
+use sp_runtime::transaction_validity::InvalidTransaction;
+use test_client::AccountKeyring;
+
+type Relayer = pallet_relayer::Module<TestStorage>;
+type Identity = pallet_identity::Module<TestStorage>;
+type AccountKeyRefCount = pallet_identity::AccountKeyRefCount<TestStorage>;
+
+type Error = pallet_relayer::Error<TestStorage>;
+
+// Relayer Test Helper functions
+// =======================================
+
+macro_rules! assert_key_usage {
+    ($user:expr, $usage:expr) => {
+        assert_eq!(AccountKeyRefCount::get(&$user.acc()), $usage);
+    };
+}
+
+#[test]
+fn basic_relayer_paying_key_test() {
+    ExtBuilder::default()
+        .monied(true)
+        .build()
+        .execute_with(&do_basic_relayer_paying_key_test);
+}
+fn do_basic_relayer_paying_key_test() {
+    let bob = User::new(AccountKeyring::Bob);
+    let alice = User::new(AccountKeyring::Alice);
+    let dave = User::new(AccountKeyring::Dave);
+
+    // add authorization for using Alice as the paying key for bob
+    assert_ok!(Relayer::set_paying_key(alice.origin(), bob.acc(),));
+
+    // No paying key is used yet.
+    assert_key_usage!(alice, 0);
+    assert_key_usage!(dave, 0);
+    assert_key_usage!(bob, 0);
+
+    // Bob accept's the paying key
+    TestStorage::set_current_identity(&bob.did);
+    let auth_id = get_last_auth_id(&Signatory::Account(bob.acc()));
+    assert_ok!(Relayer::accept_paying_key(bob.origin(), auth_id));
```

Comment:
Let's verify what's stored against expectations, including the POLYX (ideally non-zero) limit.